### PR TITLE
bug/ppb 140 fix pagination clashing with cases per page 2

### PR DESCRIPTION
**Pull request by JamesBarnes48**
## Describe your changes
Contains improvements to how we handle and validate our current page, as well as some testing and improved type coverage.

## Issue ticket number and link
https://pins-ds.atlassian.net.mcas.ms/jira/software/projects/PPB/boards/373?selectedIssue=PPB-140

### CODE REVIEW

File: packages/lib/data/database/cached-cases-client.test.js [hunk 1]
```diff
@@ -38,5 +38,31 @@ describe('cached-entra-client', () => {
...
+				const mockCache = {
+					get: mock.fn(),
+					set: mock.fn()
+				};
+				const cacheClient = new CachedCasesClient(mockClient, mockCache);
+				const requestedPage = 7;
+				const totalPages = 5;
+				const page = cacheClient.determinePage(requestedPage, totalPages);
+				assert.strictEqual(page, totalPages);
+			});
```
Comment (abel-olagoke) [inline packages/lib/data/database/cached-cases-client.test.js:65]: Could we add a test case for the scenario where there are zero cases? I think that would be useful.

**Review by abel-olagoke — COMMENTED** [review]
Everything looks good to me. Great work!
